arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    // Select the nearest feature or move the point or polyline vertex to the given screen point.
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // If a feature hasn't been selected.
+        if selectedFeature == nil {
+            selectFeature(screenPoint: screenPoint)
+        } else {
+            // Move the feature.
+            if selectedFeatureIsPolyline {
+                movePolylineVertex(mapPoint: mapPoint)
+            } else {
+                movePoint(mapPoint: mapPoint)
+            }
+        }
+    }
+    
+    func selectFeature(screenPoint: CGPoint) {
+        // Clear any previously selected features.
+        clearSelection()
+        selectedFeature = nil
+        
+        // Identify across all layers
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 10.0, returnPopupsOnly: false) { (results: [AGSIdentifyLayerResult]?, error: Error?) in
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                results?.forEach { (result) in
+                    if let featureLayer = result.layerContent as? AGSFeatureLayer {
+                        // Get a reference to the identified feature
+                        self.selectedFeature = result.geoElements[0] as? AGSFeature
+                        // If the selected feature is a polyline with any part containing more than one segment (i.e. a curve).
+                        if let polyline = self.selectedFeature?.geometry as? AGSPolyline {
+                            let polylineArray = polyline.parts.array()
+                            polylineArray.forEach { part in
+                                if part.pointCount > 2 {
+                                    // Set selected feature to nil.
+                                    self.selectedFeature = nil
+                                    // Show a message to select straight (single segment) polylines only.
+                                    self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                                    return
+                                }
+                            }
+                        }
+                        // Ensure the feature is not nil and select it.
+                        guard let selectedFeature = self.selectedFeature else { return }
+                        featureLayer.select(selectedFeature)
+                        // If the selected feature is a point, prompt the edit attributes alert.
+                        if selectedFeature.geometry?.geometryType.rawValue == 1 {
+                            self.showEditableAttributes(selectedFeature: selectedFeature)
+                        } else if selectedFeature.geometry?.geometryType.rawValue == 3 {
```

Comment:
Actually I think it's more readable if the whole block (from line 86) is a `switch case` control flow, so that you handle `selectedFeature` as polyline and point separately, rather than using `if else`.

Reply:
I could include the block starting from line 86 because it complicated things with the alert but I implemented the switch case.